### ISSUE TITLE
Result: Diffs are not compared!

### ISSUE THREAD

**Issue by Makman2** [coala/coala#4400]
Results with same values except having different diffs are considered to be equal! This is totally bad and needs to be changed.

We have to update `generate_ordering` to also take `diffs` into account (as the last entry before `debug_msg`).

This issue might involve extending tests here and in `coala-bears` to test also against diffs.

**Comment by nvzard** [comment]
@Makman2 I'd like to work on this issue :)

**Comment by nvzard** [comment]
@Makman2 I have updated  `generate_ordering` to take `diffs` into account, I have also included a test to compare two results with different diffs as `NotEqual`.
How should I extend tests in `coala-bear`?
Should I write tests for bears where I yield two results with different diffs and compare them?
Should I create another PR in `coala-bears` addressing the same issue?

**Comment by Makman2** [comment]
You don't have to test it inside coala-bears directly, but I guess some tests will fail due to this. So I would recommend you open a new issue to update coala-bears in `requirements.txt` including your commit, and adapting the tests.

**Comment by nvzard** [comment]
Done opened the issue :+1: 
https://github.com/coala/coala-bears/issues/2305
Can you please assign this to me ;)

**Comment by nvzard** [comment]
@Makman2 do I have to wait for the next release and then bump coala version in `requirements.txt` of coala-bears and modify tests accordingly OR is there any other way around?

**Comment by Makman2** [comment]
You can point the requirements to a commit hash. So once this is merged on coala's side, you change `coala~=0.12` to some git-link-like thing. I don't know the exact syntax right now, but you can google it up :+1:
